diff --git a/generic/regc_locale.c b/generic/regc_locale.c
index 6a2baad..7102a9c 100644
--- a/generic/regc_locale.c
+++ b/generic/regc_locale.c
@@ -988,12 +988,12 @@ eclass(
 }
 
 /*
- - cclass - supply cvec for a character class
+ - regcclass - supply cvec for a character class
  * Must include case counterparts on request.
- ^ static struct cvec *cclass(struct vars *, const chr *, const chr *, int);
+ ^ static struct cvec *regcclass(struct vars *, const chr *, const chr *, int);
  */
 static struct cvec *
-cclass(
+regcclass(
     struct vars *v,		/* context */
     const chr *startp,		/* where the name starts */
     const chr *endp,		/* just past the end of the name */
diff --git a/generic/regcomp.c b/generic/regcomp.c
index d828b44..74b724f 100644
--- a/generic/regcomp.c
+++ b/generic/regcomp.c
@@ -187,7 +187,7 @@ static celt element(struct vars *, const chr *, const chr *);
 static struct cvec *range(struct vars *, celt, celt, int);
 static int before(celt, celt);
 static struct cvec *eclass(struct vars *, celt, int);
-static struct cvec *cclass(struct vars *, const chr *, const chr *, int);
+static struct cvec *regcclass(struct vars *, const chr *, const chr *, int);
 static struct cvec *allcases(struct vars *, pchr);
 static int cmp(const chr *, const chr *, size_t);
 static int casecmp(const chr *, const chr *, size_t);
@@ -1546,7 +1546,7 @@ brackpart(
 	endp = scanplain(v);
 	INSIST(startp < endp, REG_ECTYPE);
 	NOERR();
-	cv = cclass(v, startp, endp, (v->cflags&REG_ICASE));
+	cv = regcclass(v, startp, endp, (v->cflags&REG_ICASE));
 	NOERR();
 	dovec(v, cv, lp, rp);
 	return;
