diff --git a/unix/configure b/unix/configure
index 87dc84d..f88fe78 100755
--- a/unix/configure
+++ b/unix/configure
@@ -8098,12 +8098,11 @@ fi
 	    ;;
 	OS/390-*)
 	    SHLIB_LD_LIBS=""
+	    DL_OBJS="tclLoadDl.o"
 	    CFLAGS_OPTIMIZE=""		# Optimizer is buggy
-
 cat >>confdefs.h <<\_ACEOF
-#define _OE_SOCKETS 1
+#define _XOPEN_SOURCE 600
 _ACEOF
-
 	    ;;
 	OSF1-1.0|OSF1-1.1|OSF1-1.2)
 	    # OSF/1 1.[012] from OSF, and derivatives, including Paragon OSF/1
