diff --git a/unix/tclUnixThrd.c b/unix/tclUnixThrd.c
index 24bc72d..f2db02a 100644
--- a/unix/tclUnixThrd.c
+++ b/unix/tclUnixThrd.c
@@ -224,7 +224,9 @@ TclpThreadCreate(
     int result;
 
     pthread_attr_init(&attr);
+    #ifndef __MVS__
     pthread_attr_setscope(&attr, PTHREAD_SCOPE_SYSTEM);
+    #endif
 
 #ifdef HAVE_PTHREAD_ATTR_SETSTACKSIZE
     if (stackSize != TCL_THREAD_STACK_DEFAULT) {
@@ -263,7 +265,7 @@ TclpThreadCreate(
 		    (void * (*)(void *))(void *)proc, (void *)clientData)) {
 	result = TCL_ERROR;
     } else {
-	*idPtr = (Tcl_ThreadId)theThread;
+	*idPtr = (Tcl_ThreadId)(theThread.__);
 	result = TCL_OK;
     }
     pthread_attr_destroy(&attr);
@@ -305,7 +307,9 @@ Tcl_JoinThread(
     int result;
     unsigned long retcode, *retcodePtr = &retcode;
 
-    result = pthread_join((pthread_t) threadId, (void**) retcodePtr);
+    pthread_t threadData;
+    threadData.__=threadId;
+    result = pthread_join(threadData, (void**) retcodePtr);
     if (state) {
 	*state = (int) retcode;
     }
@@ -365,7 +369,7 @@ Tcl_ThreadId
 Tcl_GetCurrentThread(void)
 {
 #if TCL_THREADS
-    return (Tcl_ThreadId) pthread_self();
+    return (Tcl_ThreadId) pthread_self().__;
 #else
     return (Tcl_ThreadId) 0;
 #endif
